src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+/**
+ * A super-class to Address, Email and Phone classes
+ */
+public abstract class Contact {
+
+	public final String value;
+	private boolean isPrivate;
+
+	public Contact (String Value, boolean IsPrivate) {
+		this.value = Value.trim();
+		this.isPrivate = IsPrivate;
+	}
+
+	@Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public int hashCode() {
+		return value.hashCode();
+	}
+
+	public boolean isPrivate() {
+		return isPrivate;
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| this.getClass().isInstance(other) // instance of handles nulls
```

Comment:
Very good, this ensure the class type of the two objects are the same for equality check.